src/forwarder.ts

```diff
@@ -235,6 +282,10 @@ function shouldSubscribe(
   if (typeof resource.Properties.LogGroupName !== "string") {
     return false;
   }
+  // we don't want to run the shouldSubscribe validation on step function log groups since we manually add those
+  if (resource.Properties.LogGroupName.startsWith("/aws/vendedlogs/states/")) {
```

Comment:
I don't quite understand the reason we don't follow the existing pattern here. What if the log group had already been added by the customer manually to their serverless.yaml?

Reply:
@DarcyRaynerDD During testing I found that when a [log group is defined in serverless.yml](https://github.com/DataDog/cloud-integrations-lambdas/blob/duncan-harvey/step-function-serverless-plugin/step-function-lambda-serverless-plugin-node/serverless.yml#L21-L26) it won't appear in the compiled cloudformation template at this point in the deployment (`service.provider.compiledCloudFormationTemplate?.Resources`). So the log group is instead [retrieved from the step function's loggingConfig](https://github.com/DataDog/cloud-integrations-lambdas/blob/duncan-harvey/step-function-serverless-plugin/step-function-lambda-serverless-plugin-node/serverless.yml#L36) defined in serverless.yml. This approach also covers the scenario where a customer created the log group outside of their serverless.yml and hard coded a log group arn in their loggingConfig.

This line is intended to prevent duplicate subscriptions from being added when the plugin creates a new log group for a step function since the log group created by the plugin will appear in the complied cloudformation template at this point in the deployment.

Do you have a suggestion for another approach that better fits with the existing pattern?

Reply:
Perhaps just expanding the comment a bit to explain why, while we still understand the context.

Reply:
I expanded on the comment in code and added an explanation in the description of the PR with logs demonstrating when lambda function log groups and step function log groups are created

Reply:
```
 IN: after:package:initialize
Auto instrumenting functions with Datadog
Adding Lambda Library Layers to functions
Adding Datadog Lambda Extension Layer to functions
Adding Datadog Env Vars
 IN: before:package:setupProviderConfiguration
 IN: after:package:setupProviderConfiguration
Lambda function log group created
 IN: before:package:createDeploymentArtifacts
 IN: after:package:createDeploymentArtifacts
Setting Datadog Forwarder
Subscribing step function log groups to Datadog Forwarder
Found logging config for step function duncanStepFunctionServerlessPluginLogGroupPreConfigured
Warning: Setting log level to ALL for step function duncanStepFunctionServerlessPluginLogGroupPreConfigured so traces can be generated.
Warning: Setting includeExecutionData to true for step function duncanStepFunctionServerlessPluginLogGroupPreConfigured so traces can be generated.
Warning: No logging config found for step function duncanStepFunctionServerlessPluginLogGroupCreated. Creating log group and logging to it with level ALL.
Warning: Datadog Lambda Extension and forwarder are both enabled. Only APIGateway and Step Function log groups will be subscribed to the forwarder.
Adding Plugin Version 5.20.0 tag
Adding source code integration
Skipping enabling source code integration because encrypted credentials through KMS/Secrets Manager is not supported for this integration. Please set either DATADOG_API_KEY in your environment, or set the apiKey parameter in Serverless.
 IN: before:package:compileLayers
 IN: after:package:compileLayers
 IN: before:package:compileFunctions
 IN: after:package:compileFunctions
 IN: before:package:compileEvents
 IN: after:package:compileEvents
 IN: before:package:finalize
Setting Datadog Forwarder
Subscribing step function log groups to Datadog Forwarder
**Found logging config for step function duncanStepFunctionServerlessPluginLogGroupPreConfigured
Found logging config for step function duncanStepFunctionServerlessPluginLogGroupCreated
Warning: Datadog Lambda Extension and forwarder are both enabled. Only APIGateway and Step Function log groups will be subscribed to the forwarder.**
Adding Plugin Version 5.20.0 tag
Adding source code integration
Skipping enabling source code integration because encrypted credentials through KMS/Secrets Manager is not supported for this integration. Please set either DATADOG_API_KEY in your environment, or set the apiKey parameter in Serverless.
 IN: before:aws:package:finalize:addExportNameForOutputs
 IN: after:aws:package:finalize:addExportNameForOutputs
 IN: before:aws:package:finalize:mergeCustomProviderResources

 IN: after:aws:package:finalize:mergeCustomProviderResources
**Step Function log group created**

 IN: before:aws:package:finalize:stripNullPropsFromTemplateResources
 IN: after:aws:package:finalize:stripNullPropsFromTemplateResources
 IN: before:aws:package:finalize:saveServiceState
 IN: before:aws:common:moveArtifactsToPackage:move
 IN: after:aws:common:moveArtifactsToPackage:move
 IN: after:aws:package:finalize:saveServiceState
 IN: after:package:finalize
 IN: before:aws:common:validate:validate
 IN: after:aws:common:validate:validate
```